TemplateBicep/main.bicep

```diff
@@ -0,0 +1,66 @@
+param location string = resourceGroup().location
+param uniqueSolutionPrefix string
+param discoveryZipUrl string
+
+module iotHub 'modules/iothub.bicep' = {
+  name: 'iotHub'
+  params: {
+    name: '${uniqueSolutionPrefix}hub'
+    location: location
+  }
+}
+
+module storage 'modules/storage.bicep' = {
+  name: 'storage'
+  params: {
+    storageAccountName: '${uniqueSolutionPrefix}storage'
+    storageAccountType: 'StorageV2'
+    credentialsContainerName: 'stationcredentials'
+    firmwareUpgradesContainerName: 'fwupgrades'
+    location: location
+  }
+}
+
+module redisCache 'modules/redis.bicep' = {
+  name: 'redisCache'
+  params: {
+    uniqueSolutionPrefix: uniqueSolutionPrefix
+    location: location
+  }
+}
+
+module function 'modules/function.bicep' = {
+  name: 'function'
+  params: {
+    appInsightName: observability.outputs.appInsightName
+    logAnalyticsName: observability.outputs.logAnalyticsName
+    deployDevice: true
+    uniqueSolutionPrefix: uniqueSolutionPrefix
+    useAzureMonitorOnEdge: true
+    hostingPlanLocation: location
+    redisCacheName: redisCache.outputs.redisCacheName
+    iotHubName: iotHub.outputs.iotHubName
+    storageAccountName: storage.outputs.storageAccountName
+  }
+}
+
+module observability 'modules/observability.bicep' = {
+  name: 'observability'
+  params: {
+    prefix: uniqueSolutionPrefix
+    iotHubName: iotHub.outputs.iotHubName
+    location: location
+  }
+}
+
+module discoveryService 'modules/discoveryService.bicep' = {
```

Comment:
The deployment of this service should be conditional based on param https://github.com/Azure/iotedge-lorawan-starterkit/blob/6a07c8f527e38506493f6f9ab0d5dabfe47d9ba6/Template/azuredeploy.json#L393

Reply:
Changes added. Please review.